### PR TITLE
Added `AppBuilder.start()`, combining `finish()` and `run()`

### PR DESCRIPTION
**Pull request by David-OConnor**
Added alternative starting code: `AppBuilder.start()`, which combines `.finish()` and `.run()`. These two are usually associated with each other. This flow's a little cleaner.

```rust
#[wasm_bindgen(start)]
pub fn render() {
    seed::App::build(|_, _| Init::new(Model::default()), update, view)
        .finish()
        .run();
}
```

 to 

```rust
#[wasm_bindgen(start)]
pub fn render() {
    seed::App::build(|_, _| Init::new(Model::default()), update, view)
        .start();
}
```

Are there any cases where we do something between `finish()` and `run()`? Can still do the former, but am wondering for documentation purposes.

Open to alternatives / further simplifications / renames etc. For example, what do you think about `seed::build` to start, instead of `seed::App::build`? This would be a little simpler, but hides that it's instantiating an `App`.

Overall intent is to minimize boilerplate, if we can do it with minimal drawbacks.

### CODE REVIEW

**Comment by MartinKavik** [comment]
> which combines .finish() and .run()

https://github.com/David-OConnor/seed/issues/250#issuecomment-544512869
=> `start` vs `build_and_run` - I chose `build_and_run` because `builder()...build()` is a common pattern in Rust (see the end of https://github.com/David-OConnor/seed/issues/250#issuecomment-543983556) so IDE's autocomplete can help experienced Rust programmers who know that pattern and it's a little bit more expressive imho. And when we need to split `build` and `run` phase, it would be easier + it should be a little bit less painful breaking change for current users.

> seed::build to start, instead of seed::App::build

As you said - it hides that it's `App`. And it pollutes "global" `seed::` namespace. So you are losing context and there can be problems in the future once we decide to write another important builder for some other entities.

**Comment by David-OConnor** [comment]
I agree with you on not using `seed::build`, and leaving `App` exposed.

What about #[wasm_bindgen(start)]
pub fn render() {
    seed::App::new(|_, _| Init::new(Model::default()), update, view)
        .start();
}?

I'm looking at the builder pattern as an implementation detail that allows some details (`routes`, `before_mount` etc) to be optional, and `start` (or something else) is simple, vice a formal pattern.

File: src/vdom/builder.rs [hunk 1]
```diff
@@ -183,4 +180,10 @@ impl<Ms, Mdl, ElC: View<Ms> + 'static, GMs: 'static> Builder<Ms, Mdl, ElC, GMs>
 
         app
     }
+
+    /// Start the app; wraps `.finish()` and `.start()`.
```
Comment (MartinKavik) [inline src/vdom/builder.rs]: I think that it should be:
```rust
/// Start the app; wraps `.finish()` and `.run()`.
```
But I suggest to change it to:
```rust
/// Build and run the app.
```
and mark `finish` and `run` as deprecated.

---

I was googling a little bit to make sure that we don't make mistake with `build_and_run` (vs e.g. `start`) - and I can confirm that `build` is the most used final method for builders and it's used also in https://github.com/colin-kiegel/rust-derive-builder, so I still think that it's a good idea.
`build_and_run` vs `build_and_start` - `start` sounds better for application but our old API has method `run` - I don't have preference here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,16 +3,19 @@
 ## v0.4.2
 - Added an `Init` struct, which can help with initial routing (Breaking)
 - The `routes` function now returns an `Option<Msg>` (Breaking)
-- updated `Tag::from()` to accept more input types
+- Updated `Tag::from()` to accept more input types
+- `style!` now accepts types that implement the new `ToCssValue` trait
 - Fixed a bug affecting element render order
 Added a `hashchange` listener
 - Improved error-handling
+- Tweakd bootstrap order so that `main_el_vdom` is initialized first (internal)
```
Comment (MartinKavik) [inline CHANGELOG.md]: Tweaked

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,16 +3,19 @@
 ## v0.4.2
 - Added an `Init` struct, which can help with initial routing (Breaking)
 - The `routes` function now returns an `Option<Msg>` (Breaking)
-- updated `Tag::from()` to accept more input types
+- Updated `Tag::from()` to accept more input types
+- `style!` now accepts types that implement the new `ToCssValue` trait
 - Fixed a bug affecting element render order
 Added a `hashchange` listener
```
Comment (MartinKavik) [inline CHANGELOG.md]: -

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,16 +3,19 @@
 ## v0.4.2
 - Added an `Init` struct, which can help with initial routing (Breaking)
 - The `routes` function now returns an `Option<Msg>` (Breaking)
-- updated `Tag::from()` to accept more input types
+- Updated `Tag::from()` to accept more input types
+- `style!` now accepts types that implement the new `ToCssValue` trait
```
Comment (MartinKavik) [inline CHANGELOG.md]: `ToCssValue` has been renamed to `ToCSSValueForCSSValue` and documentation is hidden.
So I suggest to change the line to something like 
```
- `style!` now accepts also `Option<impl ToString>`
```
---
I don't know if we want to add public trait `ToCSSValue` so users can implement it for custom types. If we want to discuss it I suggest to create a new issue.

File: src/vdom/builder.rs (hunk 1 above)
Comment (David-OConnor) [reply src/vdom/builder.rs]: `build_and_start` works.

**Pull request merged**